crates/load-balancer/src/strategies/mod.rs

```diff
@@ -0,0 +1,130 @@
+//! Strategies module
+//!
+mod adaptive;
+mod least_connections;
+mod round_robin;
+mod weighted_round_robin;
+
+pub use adaptive::AdaptiveStrategy;
+pub use least_connections::LeastConnectionsStrategy;
+pub use round_robin::RoundRobinStrategy;
+pub use weighted_round_robin::WeightedRoundRobinStrategy;
+
+use std::str::FromStr;
+use crate::{
+    backend::BackendId,
+    error::{Error, Result},
+};
+
+/// Strategy Trait
+/// 
+pub trait Strategy: Send + Sync + 'static {
```

Comment:
suggestion: could be `async_trait` with a bit cleaner return type for the function